dna/models.py

```diff
@@ -413,21 +404,18 @@ def __init__(
         self.lstm.to(device=self.device)
 
         lstm_output_size = hidden_state_size * n_directions
+        output_submodule_input_size = lstm_output_size + metafeatures_length
+
+        if isinstance(self, DAGRNN):
```

Comment:
This is probably not good design. A parent class has no knowledge of its children classes.